cmake/MujocoDependencies.cmake

```diff
@@ -116,18 +127,37 @@ findorfetch(
   qhull
   EXCLUDE_FROM_ALL
 )
-# MuJoCo includes a file from libqhull_r which is not exported by the qhull include directories.
-# Add it to the target.
-target_include_directories(
-  qhullstatic_r INTERFACE $<BUILD_INTERFACE:${qhull_SOURCE_DIR}/src/libqhull_r>
-)
-target_compile_options(qhullstatic_r PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
-target_link_options(qhullstatic_r PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
+
+if(NOT MUJOCO_USE_SYSTEM_qhull)
+  # MuJoCo includes a file from libqhull_r which is not exported by the qhull include directories.
+  # Add it to the target.
+  target_include_directories(
+    qhullstatic_r INTERFACE $<BUILD_INTERFACE:${qhull_SOURCE_DIR}/src/libqhull_r>
+  )
+  target_compile_options(qhullstatic_r PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
+  target_link_options(qhullstatic_r PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
+else()
```

Comment:
I am not sure of the modification/simplification you are proposing here. Feel free to suggest the specific code modification, thanks!

Reply:
The only place we use qhull is: user_mesh.cc
#include <libqhull_r/qhull_ra.h>
So I thought Qhull::qhullstatic_r should be a better target:
```
findorfetch(
  USE_SYSTEM_PACKAGE
  ${MUJOCO_LIB_USE_SYSTEM_QHULL}
  PACKAGE_NAME
  qhull
  LIBRARY_NAME
  qhull
  GIT_REPO
  https://github.com/qhull/qhull.git
  GIT_TAG
  ${MUJOCO_DEP_VERSION_qhull}
  TARGETS
--- qhull
+++ Qhull::qhullstatic_r
  EXCLUDE_FROM_ALL
)
```
But but after a little experimenting I am not sure how it will work.